### ISSUE TITLE
DH primes are couple of bits longer

### ISSUE THREAD

**Issue by karel-m** [libtom/libtomcrypt#114]
This is related to PR https://github.com/libtom/libtomcrypt/pull/111 but I think it is worth a separate issue.

As @lkinley noticed in another issue https://github.com/DCIT/perl-CryptX/issues/4 DH primes `p` in `dh_static.c` are longer than one would expect - here is the summary:

```
DH-768 real bits:784
DH-1024 real bits:1036
DH-1280 real bits:1288
DH-1536 real bits:1540
DH-1792 real bits:1792
DH-2048 real bits:2072
DH-2560 real bits:2576
DH-3072 real bits:3080
DH-4096 real bits:4116
```

By real bits I mean bits without any leading zeroes.

Was there any reason in the past why `p` values were chosen that long?

**Comment by karel-m** [comment]
My suggestion (perhaps for v2) is to drop current DH params included in libtomcrypt and use instead "battle-tested" DH params from https://tools.ietf.org/html/rfc7296#appendix-B and https://tools.ietf.org/html/rfc3526

**Comment by karel-m** [comment]
Here is my suggestion for replacing built-in DH groups:
```
 768-bit => size=96   (RFC7296 Group 1 MODP)
1024-bit => size=128  (RFC7296 Group 2 MODP)
1536-bit => size=192  (RFC3526 Group 5 MODP)
2048-bit => size=256  (RFC3526 Group 14 MODP)
3072-bit => size=384  (RFC3526 Group 15 MODP)
4096-bit => size=512  (RFC3526 Group 16 MODP)
6144-bit => size=768  (RFC3526 Group 17 MODP)
8192-bit => size=1024 (RFC3526 Group 18 MODP)
```
As DH was not part of 1.17 we are not breaking backwards compatibility here.

**Comment by karel-m** [comment]
see 1308f546bc

**Comment by karel-m** [comment]
closed by #225
